src/lib/core/mp_interval.c

```diff
@@ -99,13 +99,34 @@ interval_pack(char *data, const struct interval *itv)
 }
 
 struct interval *
-interval_unpack(const char **data, struct interval *itv)
+interval_unpack(const char **data, uint32_t len, struct interval *itv)
 {
+	/*
+	 * MsgPack extensions have length greater or equal than 1 by
+	 * specification.
+	 */
+	assert(len > 0);
+
+	const char *end = *data + len;
+	uint32_t count = mp_load_u8(data);
+	len -= sizeof(uint8_t);
+
 	memset(itv, 0, sizeof(*itv));
-	uint32_t len = mp_load_u8(data);
-	for (uint32_t i = 0; i < len; ++i) {
+	for (uint32_t i = 0; i < count; ++i) {
+		if (len < 2)
```

Comment:
Why are you checking `len` here? I mean it doesn't change anymore, so why not check it before the loop?

Reply:
I believe you can just return `itv` if `count` == 0. I think this looks better than an unnecessary check of two conditions.

Reply:
No, we must make sure that all of the provided length is used.